src/read_statmech_quantities.jl

```diff
@@ -23,41 +24,70 @@ function setup_ionization_energies(fname=joinpath(_data_dir,
 
-Returns a Dict holding the default (Barklem & Collet 2016) log equilibrium constants, which are in
-pressure units.
+# Default equilibrium constants
+Molecules have equilibrium constants in addition to patition functions.  For the diatomics, these 
+are provided by Barklem and Collet, which extensively discusses the dissociation energies.  For 
+polyatomics, we calculate these ourselves, using atomization energies calculated from the enthalpies 
+of formation at 0K from [NIST's CCCDB](https://cccbdb.nist.gov/hf0k.asp).
 
-In equation 7 in this paper, m is the reduced mass, m₁m₂/(m₁ + m₂).  This can be verified by 
-re-deriving the equilibrium constants from the partition functions and dissolution constants 
-provided by the paper.
+Korg's equilbrum constants are in terms of partial pressures, since that's what Barklem and Collet 
+provide.
 """
-setup_equilibrium_constants() = read_Barklem_Collet_table(
-                                 joinpath(_data_dir, "BarklemCollet2016-equilibrium_constants.dat"),
-                                 transform=x->x+1) #convert from log(mks) to log(cgs)
+function setup_partition_funcs_and_equilibrium_constants()
+    mol_U_path = joinpath(_data_dir, "BarklemCollet2016-molecular_partition.dat")
+    partition_funcs = merge(read_Barklem_Collet_table(mol_U_path), 
+                           load_atomic_partition_functions(),
+                           load_exomol_partition_functions())
+
+    BC_Ks = read_Barklem_Collet_table(joinpath(_data_dir, "BarklemCollet2016-equilibrium_constants.dat"),
+                                      transform=x->x+1) #convert from log(mks) to log(cgs)
+
+    # equlibrium constants for polyatomics (done here because the partition functions are used)
+    atomization_Es = CSV.File(joinpath(_data_dir, "polyatomic_partition_funcs", "atomization_energies.csv"))
+    polyatomic_Ks = map(zip(Korg.Species.(atomization_Es.spec), atomization_Es.energy)) do (spec, D00)
+        D00 *= 0.01036 # convert from kJ/mol to eV
+        # closure mapping logT to K. The let block means that default paritition functions will be 
+        # used even when different ones are passed to synthesize.  This is the desired behavior, 
+        # since the equilibrium_constants should be overridden by passing them explicitly.
+        logK = let parition_funcs = partition_funcs 
+            logT -> begin
+                Zs = get_atoms(spec)
+                Us_ratio = (prod([partition_funcs[Species(Formula(Z), 0)](logT) for Z in Zs])
+                            / partition_funcs[spec](logT))
+                masses_ratio = prod([atomic_masses[Z] for Z in Zs]) / get_mass(spec)
+                T = exp(logT)
+                translational_U_factor = (2π*kboltz_cgs*T/hplanck_cgs^2)^1.5
+                K = translational_U_factor^(length(Zs)-1) * masses_ratio^1.5 * Us_ratio * exp(-D00/(kboltz_eV*T))
+                log10(K * (kboltz_cgs*T)^(length(Zs)-1))
+            end
+        end
```

Comment:
I found these nested scopes really hard to understand.

It's also not clear to me that the `let` scope is strictly necessary. If that is indeed the case, I think something like the following is easier to understand:

```suggestion
        # closure mapping logT to K.
        function logK(logT)
            # the default paritition functions will always be used, even when different ones,
            # are passed to synthesize.  This is the desired behavior, since these 
            # equilibrium_constants should only be changed by explicitly different equilibrium
            # constants to synthesize
            dflt_parition_funcs = partition_funcs
            Zs = get_atoms(spec)
            Us_ratio = (prod([dflt_partition_funcs[Species(Formula(Z), 0)](logT) for Z in Zs])
                        / dflt_partition_funcs[spec](logT))
            masses_ratio = prod([atomic_masses[Z] for Z in Zs]) / get_mass(spec)
            T = exp(logT)
            translational_U_factor = (2π*kboltz_cgs*T/hplanck_cgs^2)^1.5
            K = translational_U_factor^(length(Zs)-1) * masses_ratio^1.5 * Us_ratio * exp(-D00/(kboltz_eV*T))
            log10(K * (kboltz_cgs*T)^(length(Zs)-1))
        end
```

Alternatively, If you really don't want to change the current structure, would you consider adding a comment after the let-statement and before the closure stating that the next line is a closure. (I found the nesting of this if statement followed by a multiline closure to be especially confusing). In this scenario, it might also help if:
- you replace `let parition_funcs = partition_funcs` with  `let deft_parition_funcs = partition_funcs`
- replace `logK` with `logK_func`

Reply:
I *think* this is  supposed to fix some performance issues with captured variables.  I will test to see if it actually makes a difference and simplify if not.  If it does I will use a name function instead of a lambda, as you suggest.

Reply:
Sorry, the comment before the let block was a thinko.  The behavior is the save whether it's there or not.  It does make a small performance difference, though, so I'm keeping it in.  I've corrected the comment.

Reply:
(I did switch the closure for a named function.)